### DISCUSSION TITLE
mailer integration

### DISCUSSION THREAD

**Discussion by gour** [ad-si/TaskLite#85] [Q&A]
Hello,

For Claws-mail and Neomutt I did provide [integration](https://claws-mail.org/faq/index.php/Using_Claws_Mail_with_other_programs#Integrating_Claws-mail_with_Taskwarrior) with TW and wonder what would be recommended way to do it with Tasklite?

One thing I'm thinking of is that for the integration I do use `taskopen` in order to be able to attach external files to the task, while Tasklite does provide simply annotation only?

**Comment by ad-si** [comment]
So far you can import emails and convert them to tasks with `tl import msg.eml` or `cat msg.eml | tl importeml`.

Simple annotations should actually be enough: If I add a filepath to a task and then print the task with `tl info <id>`, I can simply `cmd + click` on the file to open it. This feature is supported by most terminals by now. Or did you have some other use-case in mind?

Are more general file attachment where it stores the file in the database would definitely be cool as well. Doesn't have a high priority yet, but if more folks are interested I would definitely be willing to include it!

**Reply by gour** [reply]
> Or did you have some other use-case in mind?

My workflow is as follows:

1. open neomutt and check email
2.  press keyboard binding which allows me to enter task's description and which creates e.g. TW task based on some email message along with e.g. custom tag(s) and exact timestamp when the task is created (for some tasks this is very important for me)
3. by browsing TW (or Etm) I can select desired task and by e.g. using: open #taskid (via `taskopen` mechanism) it brings me back to mailer with desired email message opened (based on its msgid) 
4. after completing the task in TW, my mailer (neomutt) is opened with desired msg which can be deleted.

Does it make sense?

**Reply by gour** [reply]
> So far you can import emails and convert them to tasks with `tl import msg.eml` or `cat msg.eml | tl importeml`.

Which mailer you use to import *.eml from? Thunderbird?

**Reply by ad-si** [reply]
Cool workflow, I want that too! =D
But yeah, I'm using Thunderbird. So I'd try to find some workflow that works for Thunderbird first, but should be easy to adapt to neomutt afterwards I guess.
